src/util/heap/inspection/mod.rs

```diff
@@ -0,0 +1,52 @@
+use crate::policy::sft::SFT;
+use crate::util::linear_scan::RegionIterator;
+use crate::util::{linear_scan::Region, Address, ObjectReference};
+use crate::util::metadata::side_metadata::spec_defs::VO_BIT;
+use crate::util::heap::chunk_map::Chunk;
+
+pub trait SpaceInspector {
+    fn name(&self) -> &str;
+    fn list_regions(&self, parent_region: Option<&dyn RegionInspector>) -> Vec<Box<dyn RegionInspector>>;
+}
+
+pub(crate) fn list_child_regions<PARENT: Region, CHILD: Region + 'static>(region: &dyn RegionInspector) -> Option<Vec<Box<dyn RegionInspector>>> {
+    if region.region_type() == std::any::type_name::<PARENT>() {
+        let start_child_region = CHILD::from_aligned_address(region.start());
+        let end_child_region = CHILD::from_aligned_address(region.start() + region.size());
+        Some(RegionIterator::<CHILD>::new(start_child_region, end_child_region)
+            .map(|r| Box::new(r) as Box<dyn RegionInspector>)
+            .collect())
+    } else {
+        None
+    }
+}
+
+pub trait RegionInspector {
+    fn region_type(&self) -> &str;
```

Comment:
If something other than `<R: Region>` can implement `RegionInspector` (such as the `ImmixChunkInspector` I wrote in another comment), we can add two members:

```rust
pub trait RegionInspector {
    type RegionType: Region;
    fn get_region(): Self::RegionType;
}
```

And we can probably remove `start` and `size` because we can get the region itself.